CMakeLists.txt

```diff
@@ -223,31 +231,59 @@ if(NOT NETWORKIT_EXT_TLX)
 else()
 	add_library(tlx STATIC IMPORTED)
 	set_target_properties(tlx PROPERTIES
-			IMPORTED_LOCATION "${NETWORKIT_EXT_TLX}/lib/libtlx.a"
+			IMPORTED_LOCATION "${NETWORKIT_EXT_TLX}/lib/${CMAKE_SHARED_LIBRARY_PREFIX}tlx${CMAKE_STATIC_LIBRARY_SUFFIX}"
 			INTERFACE_INCLUDE_DIRECTORIES "${NETWORKIT_EXT_TLX}/include/")
 endif()
 
 ################################################################################
 # NETWORKIT MODULES
 if(NETWORKIT_BUILD_CORE AND NETWORKIT_MONOLITH)
+
 	if(NETWORKIT_STATIC)
 		add_library(networkit networkit/cpp/networkit.cpp)
 	else()
+		if(WIN32)
+			message(FATAL_ERROR
+					"Windows does only support static builds. Call cmake with '-DNETWORKIT_STATIC=ON'.")
+		endif()
 		add_library(networkit SHARED networkit/cpp/networkit.cpp)
 	endif()
 
+	if(WIN32 AND NETWORKIT_BUILDING_STATELIB)
+		# For Windows all global shared variables have to be exported additionally as dll
+		add_library(networkit_state SHARED networkit/cpp/GlobalState.cpp)
+		set(NETWORKIT_CXX_FLAGS "${NETWORKIT_CXX_FLAGS} /DNETWORKIT_BUILDING_STATELIB")
```

Comment:
Instead of changing `NETWORKIT_CXX_FLAGS`, we should pass this directly to `set_target_properties` (since `NETWORKIT_CXX_FLAGS` applies to all build targets).